src/poetry/core/semver/patterns.py

```diff
@@ -2,21 +2,27 @@
 
 import re
 
+from packaging.version import VERSION_PATTERN
 
-MODIFIERS = (
-    "[._-]?"
-    r"((?!post)(?:beta|b|c|pre|RC|alpha|a|patch|pl|p|dev)(?:(?:[.-]?\d+)*)?)?"
-    r"([+-]?([0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*))?"
-)
 
-_COMPLETE_VERSION = (
-    rf"v?(?:\d+!)?(\d+)(?:\.(\d+))?(?:\.(\d+))?(?:\.(\d+))?{MODIFIERS}(?:\+[^\s]+)?"
-)
+COMPLETE_VERSION = re.compile(VERSION_PATTERN, re.VERBOSE | re.IGNORECASE)
 
-COMPLETE_VERSION = re.compile("(?i)" + _COMPLETE_VERSION)
+CARET_CONSTRAINT = re.compile(
+    rf"^\^(?P<version>{VERSION_PATTERN})$", re.VERBOSE | re.IGNORECASE
```

Comment:
The `\s*` is missing here what triggered me again to question if it is required at all. I removed it in the other patterns and the tests still pass. That seems plausible because leading and trailing whitespace is handled here:
https://github.com/python-poetry/poetry-core/blob/16cb1592200694e2ea2f2dfd4931b7843a1567da/src/poetry/core/semver/helpers.py#L18

Thus, I think we can safely remove the `\s*` at the end of the other patterns.

Reply:
Sure, I'm not really fussed about it either way. Let me do that.